hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapPutIfAbsentMessageTask.java

```diff
@@ -45,6 +46,14 @@ protected ClientMessage encodeResponse(Object response) {
         return MapPutIfAbsentCodec.encodeResponse(serializationService.toData(response));
     }
 
+    @Override
+    protected void beforeProcess() {
+        super.beforeProcess();
+        if (parameters.maxIdleExist && nodeEngine.getClusterService().getClusterVersion().isLessThan(Versions.V3_11)) {
```

Comment:
this block could be extracted as a method and pulled up to `AbstractMapPutMessageTask` to avoid repetition